tests/packagers/npm.test.js

```diff
@@ -31,7 +31,7 @@ describe('npm', () => {
   });
 
   it('should return no packager sections', () => {
-    expect(npmModule.copyPackageSectionNames).toEqual([]);
+    expect(npmModule.copyPackageSectionNames).toEqual(['type']);
```

Comment:
The test name does not reflect your change, please open a new test.

Also if we are talking about ESM, shall we also copy other related sections such as `"main"`, `"exports"`?

I don't have an exhaustive list of sections to be copied, but you get the idea.

Reply:
Ok great, I can make those updates

It’s worth noting that this PR assumes that if your source is a module, you want your output to be a module too. This is not the case for many projects, where folks want to develop in ESM and output CJS for compatability. The alternate PR is a larger change, but allows users to opt in to ESM output

Reply:
AFAICT copying more sections from package.json should not affect backward compatibility, such that cjs only versions of node.js won't understand esm sections.

But you may want to take a deeper look for potential cases where newer versions of Node.js may incorrectly interpret ESM modules into CJS, this could happen for older modules. My gut feeling is not likely, I'd like your double confirm.

Reply:
Looks like the list of fields is: `type`, `exports`, `main`, and `types` (the last two are CJS fallbacks for `exports`). Sources: [node](https://nodejs.org/api/packages.html#nodejs-packagejson-field-definitions) and [ts](https://www.typescriptlang.org/docs/handbook/esm-node.html). `imports` could arguably be included too, but it's internal to a package and if you're using webpack then you shouldn't need `imports` in a build artifact

None of these are safely backwards compatible though. If I copy `type` in my existing project today, without also reworking my webpack config and a few other build tools, I hit this error in Lambda:
```
ReferenceError: require is not defined in ES module scope, you can use import instead

This file is being treated as an ES module because it has a '.js' file extension and '/var/task/package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
```

You are correct that older versions of Node would work fine, but for anyone on Node 14+, this would be breaking. And I imagine that's most users because Node 12 went EOL in 2020 with security support ending in 2022.

So if we want this to be backwards compatible, I think we should go with the other PR, and I can update the example to copy all 4 of the ESM-related fields

Reply:
Makes sense, we can close this and move to the other one.